stepman/util.go

```diff
@@ -43,6 +48,57 @@ func parseStepYml(pth, id, version string) (models.StepModel, error) {
 	return stepJSON, nil
 }
 
+// ParseStepCollection ...
+func ParseStepCollection(pth string) (models.StepCollectionModel, error) {
+	bytes, err := ioutil.ReadFile(pth)
+	if err != nil {
+		return models.StepCollectionModel{}, err
+	}
+
+	var stepCollection models.StepCollectionModel
+	if err := yaml.Unmarshal(bytes, &stepCollection); err != nil {
+		return models.StepCollectionModel{}, err
+	}
+	return stepCollection, nil
+}
+
+// DownloadStep ...
+func DownloadStep(collection models.StepCollectionModel, step models.StepModel) error {
+	downloadLocations := collection.GetdownloadLocations(step)
+
+	success := false
+	for _, downloadLocationMap := range downloadLocations {
+		for key, value := range downloadLocationMap {
+			switch key {
+			case "zip":
+				if err := DownloadAndUnZIP(value, GetStepPath(step)); err != nil {
+					log.Error("Failed to download step.zip:", err)
+					break
+				}
+				success = true
+			case "git":
+				if err := DoGitUpdate(value, GetStepPath(step)); err != nil {
+					log.Error("Failed to clone step:", err)
+					break
+				}
+				success = true
+			default:
+				log.Error("[STEPMAN] - Invalid download location")
+			}
+		}
+	}
+
+	if !success {
+		return errors.New("Failed to download step")
+	}
+	return nil
+}
+
+// GetStepPath ...
+func GetStepPath(step models.StepModel) string {
+	return GetCurrentStepCahceDir() + step.ID + "/" + step.VersionTag + "/"
```

Comment:
pls correct this method's name `GetCurrentStepCahceDir` should be: `GetCurrentStepCacheDir`